lib/output.js

```diff
@@ -1146,15 +1146,23 @@ function OutputStream(options) {
     DEFPRINT(AST_Dot, function(self, output){
         var expr = self.expression;
         expr.print(output);
-        if (expr instanceof AST_Number && expr.getValue() >= 0) {
-            if (!/[xa-f.)]/i.test(output.last())) {
-                output.print(".");
+        var prop = self.property;
+        if (RESERVED_WORDS(prop) && output.option("ie8")) {
```

Comment:
Could you put the ie8 check first? It should be faster in the non-ie8 default case.